src/VisualStudio/Core/Def/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -230,7 +234,14 @@ public FrameListener(VisualStudioActiveDocumentTracker service, IVsWindowFrame f
 
                 ((IVsWindowFrame2)frame).Advise(this, out _frameEventsCookie);
 
-                if (ErrorHandler.Succeeded(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocData, out var docData)))
+                var isDocDataInitialized = false;
+                if (ErrorHandler.Succeeded(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocCookie, out var docCookieProp)) && docCookieProp is uint docCookie)
+                {
+                    RunningDocumentInfo runningDocumentInfo = new(service._runningDocumentTable, docCookie);
+                    isDocDataInitialized = runningDocumentInfo.IsDocumentInitialized;
+                }
+
+                if (isDocDataInitialized && ErrorHandler.Succeeded(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocData, out var docData)))
```

Comment:
📝 I'm not having success here. This is going to leave `TextBuffer` permanently uninitialized, and we need this value elsewhere in code.